src/groups/mqb/mqbblp/mqbblp_queue.cpp

```diff
@@ -284,17 +196,66 @@ void Queue::dropHandleDispatched(mqbi::QueueHandle* handle, bool doDeconfigure)
     BALL_LOG_INFO << "Dropping QueueHandle [" << handle << "] for queue ["
                   << description() << "].";
 
-    if (d_localQueue_mp) {
-        doDropHandle(*d_localQueue_mp, handle, doDeconfigure);
-    }
-    else if (d_remoteQueue_mp) {
-        doDropHandle(*d_remoteQueue_mp, handle, doDeconfigure);
-    }
-    else {
-        BSLS_ASSERT_OPT(false && "Uninitialized queue");
-    }
+    // Since the handle is being dropped (which typically occurs if a client is
+    // stopping without explicitly closing its queues, or if a client crashes),
+    // we need to configure the handle's stream with null
+    // (e.g. 'consumerPriority == k_CONSUMER_PRIORITY_INVALID') parameters
+    // before we release the handle.  'configureHandle' call is eventually
+    // processed by the queue engine, which may choose to simply ignore this
+    // request if handle's stream parameters are already zero (which could
+    // occur if a client crashed immediately after sending a close-queue
+    // request).  Note that we must execute the 'configureHandle' &
+    // 'releaseHandle' operations seperately for each subStream of the
+    // 'handle'.
 
-    updateStats();
```

Comment:
Why this call is removed?
```
void QueueState::updateStats()
{
    stats()
        ->setReaderCount(handleParameters().readCount())
        .setWriterCount(handleParameters().writeCount());
}
```
Even if we close all substreams and have 0 0 reader and writer counts, we can still report this metric.